qcodes_contrib_drivers/drivers/Attocube/ANC350.py

```diff
@@ -0,0 +1,405 @@
+
+        Returns:
+            Type of the actuator
+        """
+        return self._parent.lib.get_actuator_type(dev_handle=self._parent.device_handle, axis_no=self._axis)
+
+    def _get_actuator_name(self) -> str:
+        """
+        Returns the name of the currently selected actuator
+
+        Returns:
+            Name of the actuator
+        """
+        return self._parent.lib.get_actuator_name(dev_handle=self._parent.device_handle, axis_no=self._axis)
+
+    def _get_capacitance(self) -> float:
+        """
+        Returns the motor capacitance
+        Performs a measurement of the capacitance of the piezo motor and returns the result. If no
+        motor is connected, the result will be 0.
+        The function doesn't return before the measurement is complete; this will take a few seconds
+        of time.
+
+        Returns:
+            Capacitance in Farad [F]
+        """
+        return self._parent.lib.measure_capacitance(dev_handle=self._parent.device_handle, axis_no=self._axis)
+
+    # Version 4
+    # ---------
+    def _get_voltage(self) -> float:
+        """
+        Reads back the current DC level.
+
+        Returns:
+            DC output voltage in Volts [V]
+        """
+        return self._parent.lib.get_dc_voltage(dev_handle=self._parent.device_handle, axis_no=self._axis)
+
+
+class ANC350(Instrument):
+    """
+    Qcodes driver for the ANC350
+
+    Args:
+        name: the name of the instrument itself
+        inst_num: Sequence number of the device to connect to
+        search_usb: True (default) to search for USB devices; False otherwise
+        search_tcp: True (default) to search for TCP/IP devices; False otherwise
+        library: library that fits to the version of the device and provides the appropriate dll wrappers
+
+    Attributes:
+        device_info: Returns available information about a device as a tuple.
+    """
+
+    def __init__(self, library, name: str, version: int = 3, inst_num: int = 0, search_usb: bool = True,
+                 search_tcp: bool = True):
+        super().__init__(name)
+
+        self.lib = library
+
+        self._dev_no = inst_num
+        self._version = version
+
+        if self.lib.discover(search_usb=search_usb, search_tcp=search_tcp) < inst_num:
+            raise RuntimeError("No matching device found for this number")
+        else:
+            self.device_handle = self.lib.connect(inst_num)
+
+        # TODO: is this snapshotable or should it be?
+        axischannels = ChannelList(self, "Anc350Axis", Anc350Axis)
+        for nr, axis in enumerate(['x', 'y', 'z'], 0):
+            axis_name = "{}_axis".format(axis)
+            axischannel = Anc350Axis(parent=self, name=axis_name, axis=nr)
+            axischannels.append(axischannel)
+            self.add_submodule(name, axischannel)
+        axischannels.lock()
+        self.add_submodule("axis_channels", axischannels)
+
+        self.add_parameter("device_info",
```

Comment:
I don't think that the `device_info` parameter is needed. But you should implement the function `get_idn` which returns a dictionary, like this:
```py
def get_idn(self):
        return {'vendor': 'Ithaco (DL Instruments)', 'model': '1211',
                'serial': None, 'firmware': None}
```